### DISCUSSION TITLE
Cargo.toml invalid key on {{#if integration_test}}

### DISCUSSION THREAD

**Discussion by golfingal72** [apollographql/router#5228] [Q&A]
I am building the router locally and get this error compiling: 
I am building release version 1.47.0, but have seen this in earlier versions.
How do i get around this error?

`error: invalid key
  --> ../../.cargo/git/checkouts/router-18de2f110854a520/2bb67a0/apollo-router-scaffold/templates/base/Cargo.toml:18:1
   |
18 | {{#if integration_test}}
   | ^
   |
error: invalid key
  --> ../../.cargo/git/checkouts/router-18de2f110854a520/2bb67a0/apollo-router-scaffold/templates/base/xtask/Cargo.toml:10:1
   |
10 | {{#if integration_test}}
   | ^
   |
`

**Comment by bnjjj** [comment]
Which command did you run and where in the repository ?

**Reply by golfingal72** [reply]
cargo build

I have additional plugins to the base system and I have to build a new router.

**Comment by SimonSapin** [comment]
It sounds like you’re trying to build the scaffold template without instantiating it. This template is meant to be used by the `cargo-scaffold` command to generate your initial project. Consider following the steps at https://www.apollographql.com/docs/router/customizations/custom-binary/

**Comment by ThomasJamesCrawford** [comment]
I was having the same issue

downgrading from rust `1.78` to `1.76` got rid of this error for me

**Comment by norbertstrzelecki** [comment]
it helped to bump ver of `apollo-router-scaffold = { git="https://github.com/apollographql/router.git", tag = "v1.57.1" }` in `xtask/Cargo.toml`
